Production/plugins/L2EventTupleProducer.cc

```diff
@@ -360,15 +363,15 @@ class L2EventTupleProducer : public edm::EDAnalyzer {
                   int pos = -1;
                   if(particle) {
                       pos = static_cast<int>(particle - ref_ptr);
-                      if(pos < 0 || pos >= static_cast<Long64_t>(genLepton.allParticles().size()))
+                      if(pos < 0 || pos >= static_cast<Int_t>(genLepton.allParticles().size()))
                           throw cms::Exception("L2EventTupleProducer") << "Unable to determine a gen particle index.";
                   }
                   return pos;
               };
 
               auto encodeMotherIndex = [&](const std::set<const reco_tau::gen_truth::GenParticle*>& mothers) {
```

Comment:
this defetition of `encodeMotherIndex` is outdated. Please, update it to match https://github.com/cms-tau-pog/TauMLTools/blob/prod2018_v2/Production/plugins/TauTupleProducer.cc#L348